packages/react/src/components/FileUploader/FileUploaderDropContainer.js

```diff
@@ -0,0 +1,163 @@
+import React, { useCallback, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { settings } from 'carbon-components';
+import { keys, matches } from '../../tools/key';
+import uniqueId from '../../tools/uniqueId';
+
+const { prefix } = settings;
+
+export default function FileUploaderDropContainer(props) {
+  const inputRef = useRef();
+  const {
+    accept,
+    className,
+    id,
+    disabled,
+    invalid,
+    labelText,
+    multiple,
+    name,
+    onAddFiles,
+    role,
+    tabIndex,
+    ...other
+  } = props;
+  const uid = useRef(uniqueId()).current;
+  const [isActive, setActive] = useState(false);
+  const classes = classNames(`${prefix}--file__drop-container`, {
+    [`${prefix}--file__drop-container--drag-over`]: isActive,
+    [className]: className,
+  });
+  const handleChange = useCallback(
+    evt => {
+      onAddFiles(evt, {
+        addedFiles:
+          evt.type === 'drop'
+            ? [...evt.dataTransfer.files]
+            : [...evt.target.files],
+      });
+    },
+    [onAddFiles]
+  );
+  return (
+    <div
+      className={`${prefix}--file`}
+      onDragOver={evt => {
+        evt.stopPropagation();
+        evt.preventDefault();
+        setActive(true);
+        evt.dataTransfer.dropEffect = 'copy';
+      }}
+      onDragLeave={evt => {
+        evt.stopPropagation();
+        evt.preventDefault();
+        setActive(false);
+        evt.dataTransfer.dropEffect = 'move';
+      }}
+      onDrop={evt => {
+        evt.stopPropagation();
+        evt.preventDefault();
+        setActive(false);
+        handleChange(evt);
+      }}>
+      <div className={classes}>
+        <p className={`${prefix}--file-filename`}>
+          Drag and drop files here or{' '}
```

Comment:
We need to let consumer specify this text. Also we need to consider the order of this vs. `labelText`. There are some possible ways, will come back later.

Reply:
instead of making part of the label interactive, would it be better to make the entire drop container interactive?

so the component would look like this on hover or focus, and we would be able to more easily support translation

![image](https://user-images.githubusercontent.com/8265238/60518173-9633eb80-9ca6-11e9-8d2a-534a422d91a9.png)

Reply:
Sounds plausible - @shixiedesign Any thoughts?

Reply:
@asudoh @shixiedesign as per our discussion I refactored this to make the clickable area extend to the entire drop container. we can have better support for g11n now as well